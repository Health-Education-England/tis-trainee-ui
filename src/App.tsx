import React, { Fragment } from 'react';
import ReactDOM from 'react-dom';
import './App.scss';
import Profile from './components/Profile/ProfileComponent';
import BottomNavbar from './components/navigation/BottomNavbarComponent';
import { Switch, Route, Redirect } from 'react-router-dom';
import FormR from './components/FormR';
import LogInComponent from './components/authentication/LogInComponent';
import Navbar from './components/navigation/NavbarComponent';

if (process.env.NODE_ENV !== 'production') {
  var axe = require('react-axe');
  axe(React, ReactDOM, 1000);
}

const App: React.FC = () => {
  return (
    <Fragment>
      <Navbar />
      <main>
        <Switch>
          <Route path="/formr" component={FormR} />
          <Route exact path="/profile" component={Profile} />
          <Route path="/login" component={LogInComponent} />
          <Redirect exact path="/" to="/profile" />
        </Switch>
      </main>
      <footer className="bottom-navigation-menu">
        <BottomNavbar></BottomNavbar>
      </footer>
    </Fragment>
  );
};

export default App;
