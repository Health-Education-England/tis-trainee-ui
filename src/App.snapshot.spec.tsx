import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

// All the tests are run using test runner 'jest'
// This is a snapshot test using react-test-renderer

it("renders correctly when there are no items", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
