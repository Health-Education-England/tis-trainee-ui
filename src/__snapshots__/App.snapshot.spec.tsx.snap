// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
Array [
  <header
    className="nhsuk-header header"
    role="banner"
  >
    <div
      className="nhsuk-width-container nhsuk-header__container"
    >
      <div
        className="nhsuk-header__logo nhsuk-header__logo--only"
      >
        <a
          aria-label="NHS homepage"
          className="nhsuk-header__link"
        >
          <svg
            aria-labelledby="nhsuk-logo_title"
            className="nhsuk-logo"
            focusable="false"
            role="presentation"
            viewBox="0 0 40 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="nhsuk-logo_title"
            >
              NHS Logo
            </title>
            <path
              className="nhsuk-logo__background"
              d="M0 0h40v16H0z"
            />
            <path
              className="nhsuk-logo__text"
              d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"
            />
            <img
              alt="NHS Logo"
              src="https://assets.nhs.uk/images/nhs-logo.png"
            />
          </svg>
        </a>
      </div>
    </div>
  </header>,
  <main
    className="nhsuk-main-wrapper"
    id="maincontent"
  >
    <div
      className="Section__container___3YYTG"
      style={Object {}}
    />
  </main>,
  <footer>
    <div
      className="nhsuk-footer"
    >
      <div
        className="nhsuk-width-container"
      >
        <h2
          className="nhsuk-u-visually-hidden"
        >
          Support links
        </h2>
        <ul
          className="nhsuk-footer__list"
        >
          <li
            className="nhsuk-footer__list-item"
          >
            <a
              className="nhsuk-footer__list-item-link refLink"
              href="http://jobs.hee.nhs.uk"
            >
              Jobs at HEE
            </a>
          </li>
          <li
            className="nhsuk-footer__list-item"
          >
            <a
              className="nhsuk-footer__list-item-link refLink"
              href="/about/contact-us"
            >
              Contact us
            </a>
          </li>
          <li
            className="nhsuk-footer__list-item"
          >
            <a
              className="nhsuk-footer__list-item-link refLink"
              href="/about/privacy-notice"
            >
              Privacy policy
            </a>
          </li>
          <li
            className="nhsuk-footer__list-item"
          >
            <a
              className="nhsuk-footer__list-item-link refLink"
              href="/about/terms-conditions"
            >
              Terms and conditions
            </a>
          </li>
          <li
            className="nhsuk-footer__list-item"
          >
            <a
              className="nhsuk-footer__list-item-link refLink"
              href="/about/cookies"
            >
              Cookie policy
            </a>
          </li>
        </ul>
        <p
          className="nhsuk-footer__copyright"
        >
          See our
           
          <a
            href="https://twitter.com/NHS_HealthEdEng/lists/local-offices/members"
          >
            local offices
          </a>
           
          on Twitter.
        </p>
      </div>
    </div>
  </footer>,
]
`;
